### PR TITLE
reference/performance: update optimizer-hints

### PR DESCRIPTION
**Pull request by lawyerphx**
Add "ignore_plan_cache" hint for prepare statements to ignore plan cache.

### What is changed, added or deleted? (Required)

optimizer-hints.md is changed.

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version, including v4.0 changes for now)

### What is the related PR or file link(s)?

- Other reference link(s): issue [issue 15587](https://github.com/pingcap/tidb/issues/15587)

### CODE REVIEW

File: reference/performance/optimizer-hints.md [hunk 1]
```diff
@@ -265,4 +265,16 @@ select /*+ MEMORY_QUOTA(1024 MB) */ * from t;
 select /*+ READ_FROM_REPLICA() */ * from t;
 ```
 
+### IGNORE_PLAN_CACHE()
+
+`IGNORE_PLAN_CACHE()` 提示优化器处理当前 prepare 语句时不使用 plan cache。
```
Comment (TomShawn) [inline reference/performance/optimizer-hints.md]:
```suggestion
`IGNORE_PLAN_CACHE()` 提示优化器处理当前 `prepare` 语句时不使用 plan cache。
```

File: reference/performance/optimizer-hints.md [hunk 2]
```diff
@@ -265,4 +265,16 @@ select /*+ MEMORY_QUOTA(1024 MB) */ * from t;
...
+
+下面的例子会强制 prepare 语句不使用 plan cache：
```
Comment (TomShawn) [inline reference/performance/optimizer-hints.md]:
```suggestion
以下示例强制 `prepare` 语句不使用 plan cache：
```

**Comment by lawyerphx** [comment]
@eurekaka PTAL ~

File: reference/performance/optimizer-hints.md [hunk 3]
```diff
@@ -265,4 +265,16 @@ select /*+ MEMORY_QUOTA(1024 MB) */ * from t;
...
+### IGNORE_PLAN_CACHE()
+
+`IGNORE_PLAN_CACHE()` 提示优化器处理当前 prepare 语句时不使用 plan cache。
+
+以下示例强制 `prepare` 语句不使用 plan cache：
+
+{{< copyable "sql" >}}
+
+```sql
+prepare stmt from 'select  /*+ IGNORE_PLAN_CACHE() */ * from t where t.id=?';
```
Comment (eurekaka) [inline reference/performance/optimizer-hints.md]:
```suggestion
prepare stmt from 'select  /*+ IGNORE_PLAN_CACHE() */ * from t where t.id = ?';
```

File: reference/performance/optimizer-hints.md (hunk 1 above)
Comment (eurekaka) [inline reference/performance/optimizer-hints.md]: 表明一下这个 hint 一般用在 prepare-plan-cache 开启的场景下？可以引用一下 https://pingcap.com/docs-cn/dev/reference/configuration/tidb-server/configuration-file/#prepared-plan-cache 这个配置项的描述

Comment (lawyerphx) [reply reference/performance/optimizer-hints.md]: done

File: reference/performance/optimizer-hints.md [hunk 4]
```diff
@@ -266,3 +266,17 @@ select /*+ READ_FROM_REPLICA() */ * from t;
 ```
 
 除了 Hint 外，环境变量 `tidb_replica_read` 设为 `'follower'` 或者 `'leader'` 也能决定是否开启该特性。
+
+### IGNORE_PLAN_CACHE()
+
+`IGNORE_PLAN_CACHE()` 提示优化器在处理当前 prepare 语句时不使用 plan cache。
+
+该 Hint 用于在 [prepare-plan-cache](https://pingcap.com/docs-cn/dev/reference/configuration/tidb-server/configuration-file/#prepared-plan-cache) 开启的场景下临时禁用 plan cache。
```
Comment (TomShawn) [inline reference/performance/optimizer-hints.md]:
```suggestion
该 Hint 用于在 [prepare-plan-cache](/reference/configuration/tidb-server/configuration-file/#prepared-plan-cache) 开启的场景下临时禁用 plan cache。
```

File: reference/performance/optimizer-hints.md [hunk 5]
```diff
@@ -266,3 +266,17 @@ select /*+ READ_FROM_REPLICA() */ * from t;
 ```
 
 除了 Hint 外，环境变量 `tidb_replica_read` 设为 `'follower'` 或者 `'leader'` 也能决定是否开启该特性。
+
+### IGNORE_PLAN_CACHE()
+
+`IGNORE_PLAN_CACHE()` 提示优化器在处理当前 prepare 语句时不使用 plan cache。
```
Comment (TomShawn) [inline reference/performance/optimizer-hints.md]:
```suggestion
`IGNORE_PLAN_CACHE()` 提示优化器在处理当前 `prepare` 语句时不使用 plan cache。
```

**Comment by lawyerphx** [comment]
> Rest LGTM

Done now, merge it?

**Pull request merged**